src/main.rs

```diff
@@ -25,6 +39,43 @@ impl BFInterp {
         }
     }
 
+    fn store_gl_var(&mut self, module: &Module) {
+        for gl_var in module.global_vars.iter() {
+            let mut var_string = String::new();
+            let mut ignore = false;
+            match &gl_var.initializer {
```

Comment:
Right, this is a beast. ;-) Took me a while to understand what's going on. Now I'm wondering if can do the formatting in one go in our hard-coded `printf` method. So instead of dissecting the format string in here, we just store it as is in `gl_vars`. Then when the time comes to print it, we can iterate over it and construct a new string from it while at the same time replacing any `%` patterns with the appropriate values. I think that would make this code and the `prinftf` function below a bit easier to understand and is possibly more performant since we don't call `replacen` multiple times on the format string. Thoughts?

Reply:
So in essence I'm imagining something like this:
```
fn printf() {
    i = 0;
    while i < string.len() {
        if string[i] = '%' {
            newstring.push(next_arg);
            i += 2;
            continue;
        }
        string.push(string[i]);
    }
}
```

Reply:
You can them maybe add another function `arg_to_string`, that takes an argument and returns it as a string, so you don't need to do everything in one function.